lib/mblinks.js

```diff
@@ -100,7 +130,13 @@ var MBLinks = function (cachekey, expiration) {
         insert_func(mblinks.createMusicBrainzLink(mb_url, _type));
       });
     } else {
-      mblinks.ajax_requests.push($.proxy(function () {
+      // Merge with previous context if there's already a pending ajax request
+      var handlers = [];
+      if (url in mblinks.ajax_requests) {
+        handlers = mblinks.ajax_requests[url].context.handlers;
+      }
+      handlers.push(insert_func);
+      mblinks.ajax_requests.push(url, function () {
```

Comment:
I agree for the proposed key, that makes sense.
Sorry for not responding sooner, I don't have much free time these days...
Feel free to make the modification if you want.